modules/typesniffer/typesniffer.go

```diff
@@ -70,6 +72,11 @@ func (ct SniffedType) IsRepresentableAsText() bool {
 	return ct.IsText() || ct.IsSvgImage()
 }
 
+// Mime return the mime
+func (ct SniffedType) Mime() string {
+	return strings.Split(ct.contentType, ";")[0]
+}
+
```

Comment:
Note to self, this is a better version than I have in https://github.com/go-gitea/gitea/pull/20464, will incorporate there.

Reply:
we should use SplitN so go can skip split after first ; ... just some optimazion nits

Reply:
@silverwind I'm going to delete the mime stuff as it's unrelated to the pull topic! - so feel free to pick it for your pull :)

Reply:
If we are talking about micro-optimization, `str[:strings.Index(str, ";")]` should be even faster than SplitN I think :)